### PR TITLE
docs and rubocop for several post/windows/gather modules

### PR DESCRIPTION
**Pull request by h00die**
This adds docs and rubocop for:

* modules/post/windows/gather/dnscache_dump.rb 
* modules/post/windows/gather/enum_hostfile.rb
* modules/post/windows/gather/lsa_secrets.rb
* (docs only) post/windows/gather/credentials/windows_autologin.md
* (docs only) post/windows/gather/enum_ad_groups.md
* (docs only) post/windows/gather/enum_domain.md
* (docs only) post/windows/gather/enum_domain_group_users.md 

modules/post/windows/gather/lsa_secrets.rb also now stores the secrets, optionally, in the database as creds.

windows/gather/enum_domain_group_users.rb had nil loop bug if no users were in the group, so I fixed that crash.

### CODE REVIEW

File: documentation/modules/post/windows/gather/enum_hostfile.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Vulnerable Application
```
Comment (adfoster-r7) [inline documentation/modules/post/windows/gather/enum_hostfile.rb:1]: Just confirming if this is meant to be a `.rb` file? 👀

Comment (h00die) [reply documentation/modules/post/windows/gather/enum_hostfile.rb]: oops!

File: modules/post/windows/gather/enum_domain_group_users.rb [hunk 2]
```diff
@@ -73,6 +73,7 @@ def get_members(results)
 
     # Usernames start somewhere around line 6
     results = results.slice(6, results.length)
+    return members if results.nil?
```
Comment (adfoster-r7) [inline modules/post/windows/gather/enum_domain_group_users.rb:76]: :+1:

**Comment by adfoster-r7** [comment]
## Release Notes

Adds documentation and code quality changes to the following modules:

- modules/post/windows/gather/dnscache_dump.rb
- modules/post/windows/gather/enum_hostfile.rb
- modules/post/windows/gather/lsa_secrets.rb
- (docs only) post/windows/gather/credentials/windows_autologin.md
- (docs only) post/windows/gather/enum_ad_groups.md
- (docs only) post/windows/gather/enum_domain.md
- (docs only) post/windows/gather/enum_domain_group_users.md

**Comment by pbarry-r7** [comment]
# Release Notes

Updated module code and added module docs for a number of Windows <code>post</code> modules.

**Pull request merged**